src/core/opamFilename.ml

```diff
@@ -473,6 +479,13 @@ let of_json = function
   | `String x -> (try Some (of_string x) with _ -> None)
   | _ -> None
 
+let compare f g =
+  let dir = Dir.compare f.dirname g.dirname in
+  if dir <> 0 then dir else
+    Base.compare f.basename g.basename
```

Comment:
```suggestion
let compare {dirname; basename} g =
  let dir = Dir.compare dirname g.dirname in
  if dir <> 0 then dir else
    Base.compare basename g.basename
```
This would be safer in case the content of record changes. This way if one field is added we get a warning here.

Reply:
good point!